airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/source.py

```diff
@@ -0,0 +1,114 @@
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+from typing import Any, List, Mapping, Tuple
+
+from airbyte_cdk.logger import AirbyteLogger
+from airbyte_cdk.models import ConnectorSpecification
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http.auth import Oauth2Authenticator
+
+from .common import SourceContext
+from .spec import Spec
+from .streams import (
+    DisplayReportStream,
+    Profiles,
+    SponsoredBrandsAdGroups,
+    SponsoredBrandsCampaigns,
+    SponsoredBrandsKeywords,
+    SponsoredBrandsReportStream,
+    SponsoredDisplayAdGroups,
+    SponsoredDisplayCampaigns,
+    SponsoredDisplayProductAds,
+    SponsoredDisplayTargetings,
+    SponsoredProductAdGroups,
+    SponsoredProductAds,
+    SponsoredProductCampaigns,
+    SponsoredProductKeywords,
+    SponsoredProductNegativeKeywords,
+    SponsoredProductsReportStream,
+    SponsoredProductTargetings,
+)
+
+TOKEN_URL = "https://api.amazon.com/auth/o2/token"
+
+
+# Source
+class SourceAmazonAds(AbstractSource):
+    def __init__(self):
+        super().__init__()
+        self.ctx = SourceContext()
+
+    def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, any]:
+        """
+        :param config:  the user-input config object conforming to the connector's spec.json
+        :param logger:  logger object
+        :return Tuple[bool, any]: (True, None) if the input config can be used to connect to the API successfully, (False, error) otherwise.
+        """
+        config = Spec(**config)
+        Profiles(config, authenticator=self._make_authenticator(config)).fill_context()
+        return True, None
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        """
+        :param config: A Mapping of the user input configuration as defined in the connector spec.
+        """
+        config = Spec(**config)
+        auth = self._make_authenticator(config)
+        profiles_stream = Profiles(config, context=self.ctx, authenticator=auth)
+        profiles_stream.fill_context()
+        return [
+            profiles_stream,
+            SponsoredDisplayCampaigns(config, context=self.ctx, authenticator=auth),
```

Comment:
nitpicking, I think all of them has same signature, you can have list of stream classes and then instantiate them in the loop

Reply:
done